src/services/identity/DeploymentsService.ts

```diff
@@ -115,6 +118,83 @@ class DeploymentsService {
           .map(() => amplifyInfo)
       })
   }
+
+  getDeploymentInfoFromSiteId = async (repoId: string) => {
+    const deploymentInfo = await this.repository.findOne({
+      where: {
+        siteId: repoId,
+      },
+    })
+    if (!deploymentInfo)
+      return errAsync(new NotFoundError("Site has not been deployed!"))
+    return okAsync(deploymentInfo)
+  }
+
+  updateAmplifyPassword = async (
+    repoName: string,
+    encryptedPassword: string,
+    iv: string,
+    enablePassword: boolean
+  ) => {
+    const deploymentInfo = await this.repository.findOne({
+      include: [
+        {
+          model: Site,
+          required: true,
+          include: [
+            {
+              model: Repo,
+              required: true,
+              where: {
+                name: repoName,
+              },
+            },
+          ],
+        },
+      ],
+    })
+    if (!deploymentInfo)
+      return errAsync(`Deployment for ${repoName} does not exist`)
+    const { id, hostingId: appId } = deploymentInfo
+    let updateAppInput
+    if (!enablePassword) {
```

Comment:
clarification, why is this called `enablePassword` ah? wouldnt this function only be called iff the repo has enabled password in the first place? 
do you mean `defaultPassword`?

Reply:
or is this trying to __**remove**__ existing password?

Reply:
Hmm it's meant to be a boolean indicating whether the site contains a password or not - i'm open to what we can rename this to make this clearer!